src/pages/Compare/index.spec.tsx

```diff
@@ -435,63 +445,148 @@ describe(__filename, () => {
+
+    // Once the default file is loaded (without `path` defined), there is an
+    // update with `path` being set to the default file.
+    root.setProps({ path: version.file.selected_file });
+
+    expect(dispatchSpy).not.toHaveBeenCalled();
+  });
+
+  it('dispatches viewVersionFile() on mount if the query string contains a `path` that is different than `version.selectedPath`', () => {
+    const path = 'a/different/file.js';
+    const history = createFakeHistory({
+      location: createFakeLocation({
+        search: queryString.stringify({ path }),
+      }),
+    });
+    const {
+      _viewVersionFile,
+      dispatchSpy,
+      fakeThunk,
+      version,
+    } = loadDiffAndRender({ history });
+
+    expect(dispatchSpy).toHaveBeenCalledWith(fakeThunk.thunk);
+    expect(dispatchSpy).toHaveBeenCalledTimes(1);
+    expect(_viewVersionFile).toHaveBeenCalledWith({
+      versionId: version.id,
+      selectedPath: path,
+      preserveHash: true,
+    });
+  });
+
+  it('does not dispatch viewVersionFile() on mount when `path` is equal to the selected path', () => {
+    const version = fakeVersionWithDiff;
+    const history = createFakeHistory({
+      location: createFakeLocation({
+        search: queryString.stringify({ path: version.file.selected_file }),
+      }),
+    });
+
+    const { dispatchSpy } = loadDiffAndRender({ history });
+
+    expect(dispatchSpy).not.toHaveBeenCalled();
+  });
+
+  it('dispatches fetchDiff() with the path specified in the URL on mount', () => {
+    const addonId = 9999;
     const baseVersionId = 1;
     const headVersionId = baseVersionId + 1;
-    const params = getRouteParams({ addonId, baseVersionId, headVersionId });
+    const path = 'a/different/file.js';
+    const history = createFakeHistory({
+      location: createFakeLocation({
+        search: queryString.stringify({ path }),
+      }),
+    });
 
     const store = configureStore();
-    const fakeThunk = createFakeThunk();
     const dispatch = spyOn(store, 'dispatch');
+    const fakeThunk = createFakeThunk();
+    const _fetchDiff = fakeThunk.createThunk;
 
-    const root = render({
-      ...params,
-      _fetchDiff: fakeThunk.createThunk,
+    render({
+      ...getRouteParams({ addonId, baseVersionId, headVersionId }),
+      _fetchDiff,
+      history,
       store,
     });
-    // Once the default file is loaded (without `path` defined), there is an
-    // update with `path` being set to the default file.
-    root.setProps({ path: fakeVersionWithDiff.file.selected_file });
 
     expect(dispatch).toHaveBeenCalledWith(fakeThunk.thunk);
-    expect(dispatch).toHaveBeenCalledTimes(1);
+    expect(_fetchDiff).toHaveBeenCalledWith({
+      addonId,
+      baseVersionId,
+      headVersionId,
+      path,
+    });
+  });
+
+  // This could happen when a keyboard navigation updates the selected path. We
+  // do not want to update the selected path again, so we apply this logic to
+  // the first render (mount) only.
```

Comment:
This comment is helpful but I think it should move to `loadData()` where the logic happens.